docs/articles/infrastructure/ci-cd/yocto-cache.rst

```diff
@@ -0,0 +1,101 @@
+Yocto cache
+===========
+
+This chapter explains how to set up a cache for use with yocto. While building
+an image, yocto caches everything it builds and downloads. This is useful since
+it enables incremental builds where one only has to re-build what was changed
+since last build, but also this makes it possible to share this cache between
+builds. When used properly, the cache will dramatically improve build times,
+usually up to 50% faster.
+
+Specifically, we will be talking about the ``downloads`` directory and the
+``sstate-cache`` directory in this chapter. The ``downloads`` directory contains
+all downloaded packages, including those downloaded over svn or git or similar,
+and ``sstate-cache`` contains everything that was built, so basically binaries
+plus some meta-data stating that it has actually been built and what version and
+so on. One reason why the meta-data is useful is because otherwise the system
+might try to reuse binaries for x86 when building for armv7, which wouldn't
+work.
+
+In general, when using a cached build in yocto, there are two options that needs
+to be set: ``SOURCE_MIRROR_URL`` and ``SSTATE_MIRRORS``. The first one points to
+where source packages can be found, the second points to the cached binary
+builds.
+
+.. code-block: bash
+    :caption: site.conf
+
+    INHERIT += "own-mirrors"
+    SOURCE_MIRROR_URL ?= "http://example.org/yocto-cache/downloads/"
+    SSTATE_MIRRORS ?= "file://.* http://example.org/yocto-cache/sstate-cache/PATH"
```

Comment:
What does the `file://.*` do here?

Reply:
It's from the guide on how to set these things up actually. Not sure it is needed actually, but just copied from the existing `site.conf` in pelux-manifests.

Reply:
Ah yeah, I remember now putting it there and when I tested without then it wouldn't work for some reason.